DSharpPlus/Net/Rest/DiscordApiClient.cs

```diff
@@ -769,6 +770,157 @@ internal async Task UpdateUserVoiceStateAsync(ulong guild_id, ulong user_id, ulo
         }
         #endregion
 
+        #region Stickers
+
+        internal async Task<DiscordMessageSticker> GetGuildStickerAsync(ulong sticker_id, ulong guild_id)
+        {
+            var route = $"{Endpoints.GUILDS}/:guild_id{Endpoints.STICKERS}/:sticker_id";
+            var bucket = this.Rest.GetBucket(RestRequestMethod.GET, route, new {guild_id, sticker_id}, out var path);
+            var url = Utilities.GetApiUriFor(path);
+
+            var res = await this.DoRequestAsync(this.Discord, bucket, url, RestRequestMethod.GET, route).ConfigureAwait(false);
+            var json = JObject.Parse(res.Response);
+            var ret = json.ToDiscordObject<DiscordMessageSticker>();
+            if (json["user"] is not null) // Null = Missing stickers perm //
+            {
+                var tsr = json["user"].ToDiscordObject<TransportUser>();
+                var usr = new DiscordUser(tsr) {Discord = this.Discord};
+                usr = this.Discord.UserCache.AddOrUpdate(tsr.Id, usr, (id, old) =>
+                {
+                    old.Username = usr.Username;
+                    old.Discriminator = usr.Discriminator;
+                    old.AvatarHash = usr.AvatarHash;
+                    return old;
+                });
+                ret.User = usr;
+            }
```

Comment:
I know we cache things throughout this class but I've recently decided that this is bad practice, so I think this should be removed. This is a design flaw on our part and we should only be caching things on the gateway side. This is how it will be in the rewrite and I will go through this class at some point to remove caching on the rest side.